bluefog/common/mpi_context.cc

```diff
@@ -310,13 +313,31 @@ void MPIContext::Initialize(const std::vector<int>& ranks,
     BFLOG(DEBUG) << "Using the existing mpi_comm.";
   }
 
-  // Create local comm, Determine local rank by querying the local communicator.
-  MPI_Comm_split_type(mpi_comm, MPI_COMM_TYPE_SHARED, 0, MPI_INFO_NULL,
-                      &local_comm);
+  int world_rank, world_size;
+  MPI_Comm_rank(mpi_comm, &world_rank);
+  MPI_Comm_size(mpi_comm, &world_size);
+
+  const char* nodes_per_machine_ptr = std::getenv(BLUEFOG_NODES_PER_MACHINE);
+  if (nodes_per_machine_ptr == nullptr) {
+    // Create local comm, Determine local rank by querying the local
+    // communicator.
+    MPI_Comm_split_type(mpi_comm, MPI_COMM_TYPE_SHARED, 0, MPI_INFO_NULL,
+                        &local_comm);
+  } else {
+    const int local_size = atoi(nodes_per_machine_ptr);
+    if (local_size <= 0) {
+      BFLOG(FATAL) << "The provided NODES_PER_MACHINE must be positive integer.";
```

Comment:
Does FATAL here stops the program from running?
If not, the rest lines will still be executed.

Reply:
Yes, FATAL will stop the program. (ERROR will not).